### PR TITLE
Add Wishlists authorization logic and index action

### LINKED ISSUE #11: [Dependent on Issues #8, #6] Add Rule - Only Signed in user can see list of his wishlists
Set a rule that only logged in user can see a list of his wishlists.

### LINKED ISSUE #12: [Dependent on Issue #8, #6] Add Rule - Only Signed in user can create a wishlist
Set a rule that only logged in user can create a wishlist

### LINKED ISSUE #13: [Dependent on Issues #8, #6] Add Rule - Only Signed in user can edit a wishlist
URI /wishlists/:id/edit
wishlists#edit

### LINKED ISSUE #24: View 'list of wishlists'
On the main page user can see the list of all wishlists he made.

### PR DESCRIPTION
**Pull request by student-tomasz**
@agalkiewicz @nodzy I've stolen some issues from you guys, so that the functionality that we already had on `master` could remain intact after adding the authorization logic. Sorry for that.

Closes #24 
Closes #11 
Closes #12

### CODE REVIEW

File: spec/controllers/wishlists_controller_spec.rb [hunk 1]
```diff
@@ -1,84 +1,124 @@
...
+
+        it "returns 200 HTTP status code" do
+          expect(response).to have_http_status(200)
+        end
+
+        it "assigns a newly created but unsaved wishlist as @wishlist" do
+          expect(assigns(:wishlist)).to be_a_new(Wishlist)
+        end
+
+        it "doesn't create new wishlist" do
```
Comment (agalkiewicz) [inline spec/controllers/wishlists_controller_spec.rb]: It's a little bit confusing title comparing it with one above. Maybe sth like "doesn't save new wishlist to a database"?

File: spec/controllers/wishlists_controller_spec.rb [hunk 2]
```diff
@@ -1,84 +1,124 @@
...
+        end
       end
     end
+  end
 
-    context "with invalid parameters" do
-      before do
-        request_post(invalid_parameters)
+  context "when not logged in" do
+    shared_examples "redirects to the 'sign_in' page" do
```
Comment (agalkiewicz) [inline spec/controllers/wishlists_controller_spec.rb:81]: Nice one. I will use it.

File: app/views/wishlists/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<h1>Wishlists</h1>
+<div>
+  <ul>
+    <%- @wishlists.each do |wishlist| %>
+    <li>
```
Comment (agalkiewicz) [inline app/views/wishlists/index.html.erb]: Incorrect indentation?

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
-  resources :wishlists, only: [:new, :create]
+  root to: "wishlists#index"
+  resources :wishlists, only: [:index, :new, :create]
```
Comment (agalkiewicz) [inline config/routes.rb:3]: You can also write `%i(index new create)`.

File: app/views/wishlists/index.html.erb (hunk 3 above)
Comment (student-tomasz) [inline app/views/wishlists/index.html.erb]: Oops.

File: config/routes.rb (hunk 4 above)
Comment (student-tomasz) [inline config/routes.rb:3]: Good point, but I like the literal syntax better. Don't want to change. ;)

Comment (agalkiewicz) [inline config/routes.rb:3]: 3 characters less. :P I think Oskar will point it. :P

**Comment by agalkiewicz** [comment]
:+1:

File: app/views/wishlists/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<h1>Wishlists</h1>
+<div>
+  <ul>
+    <%- @wishlists.each do |wishlist| %>
+      <li>
+        <%# link_to wishlist.name, edit_wishlist_path -%>
```
Comment (Kacper3331) [inline app/views/wishlists/index.html.erb]: Maybe remove this commented line?

**Comment by Kacper3331** [comment]
👍

**Comment by Kacper3331** [comment]
I think it's okay :+1:

File: app/views/wishlists/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<h1>Wishlists</h1>
+<div>
+  <ul>
+    <%- @wishlists.each do |wishlist| %>
```
Comment (agalkiewicz) [inline app/views/wishlists/index.html.erb]: `<%-` - typo or it means sth?

Comment (student-tomasz) [inline app/views/wishlists/index.html.erb]: Typo, the `-` here doesn't mean anything, strange it worked.

File: config/routes.rb (hunk 4 above)
Comment (gotar) [inline config/routes.rb:3]: :trollface: 
 it's ok

File: spec/controllers/wishlists_controller_spec.rb [hunk 7]
```diff
@@ -1,83 +1,123 @@
 require "rails_helper"
 
 RSpec.describe WishlistsController do
-  let(:valid_parameters) { attributes_for(:wishlist, user_id: create(:user).id) }
+  context "when user is logged in" do
+    sign_in_user
```
Comment (gotar) [inline spec/controllers/wishlists_controller_spec.rb]: this should be in before block

**Comment by gotar** [comment]
looks ok

File: spec/controllers/wishlists_controller_spec.rb (hunk 7 above)
Comment (student-tomasz) [inline spec/controllers/wishlists_controller_spec.rb]: It's a method that gets extended into each controller spec and invokes a `before` block defined in it. But it's a not obvious behavior, so I've updated it to be included and invoked on normal terms in explicit `before` blocks.

**Pull request merged**